File: example_services/scala/finatra/example/example-persistence/src/main/scala/com/me/example/persistence/Dao.scala
package com.me.example.persistence

import com.twitter.util.Future

trait Dao[K, V] {
  def get(key: K): Future[Option[V]]

  def set(key: K, data: V): Future[Unit]

  def remove(key: K): Future[Unit]

  def update(key: K, value: V): Future[Unit]
}

File: example_services/java/finatra/example/example-persistence/src/main/java/com/me/example/persistence/Dao.java
package com.me.example.persistence;

import java.util.Optional;

public interface Dao<K,V> {
    abstract public Optional<V> get(K key);
    abstract public void set(K key, V data);
    abstract public void remove(K key);
    abstract public void update(K key, V value);
}
